tools/src/test/java/org/infinispan/tools/jdbc/migrator/marshaller/LegacyVersionAwareMarshallerTest.java

```diff
@@ -0,0 +1,171 @@
+
+import static org.testng.Assert.assertEquals;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.infinispan.commons.io.ByteBufferImpl;
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.commons.marshall.MarshallUtil;
+import org.infinispan.commons.marshall.StreamingMarshaller;
+import org.infinispan.container.entries.ImmortalCacheEntry;
+import org.infinispan.container.entries.ImmortalCacheValue;
+import org.infinispan.container.entries.MortalCacheEntry;
+import org.infinispan.container.entries.MortalCacheValue;
+import org.infinispan.container.entries.TransientCacheEntry;
+import org.infinispan.container.entries.TransientCacheValue;
+import org.infinispan.container.entries.TransientMortalCacheEntry;
+import org.infinispan.container.entries.TransientMortalCacheValue;
+import org.infinispan.container.entries.metadata.MetadataImmortalCacheEntry;
+import org.infinispan.container.entries.metadata.MetadataImmortalCacheValue;
+import org.infinispan.container.entries.metadata.MetadataMortalCacheEntry;
+import org.infinispan.container.entries.metadata.MetadataMortalCacheValue;
+import org.infinispan.container.entries.metadata.MetadataTransientCacheEntry;
+import org.infinispan.container.entries.metadata.MetadataTransientCacheValue;
+import org.infinispan.container.entries.metadata.MetadataTransientMortalCacheEntry;
+import org.infinispan.container.entries.metadata.MetadataTransientMortalCacheValue;
+import org.infinispan.container.versioning.NumericVersion;
+import org.infinispan.container.versioning.SimpleClusteredVersion;
+import org.infinispan.metadata.EmbeddedMetadata;
+import org.infinispan.metadata.Metadata;
+import org.infinispan.metadata.impl.InternalMetadataImpl;
+import org.infinispan.test.data.Key;
+import org.infinispan.test.data.Person;
+import org.infinispan.util.KeyValuePair;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+/**
+ * Tests to ensure that LegacyVersionAwareMarshaller can correctly unmarshall infinispan 8.x bytes.
+ */
+@Test(testName = "tools.LegacyVersionAwareMarshallerTest", groups = "functional")
+public class LegacyVersionAwareMarshallerTest {
+
+   private final StreamingMarshaller marshaller;
+   private Map<String, byte[]> byteMap;
+
+   public LegacyVersionAwareMarshallerTest() {
+      Map<Integer, AdvancedExternalizer<?>> externalizerMap = new HashMap<>();
+      externalizerMap.put(256, new TestObjectExternalizer());
+      marshaller = new LegacyVersionAwareMarshaller(externalizerMap);
+   }
+
+   @BeforeTest
+   public void beforeTest() throws Exception {
+      Path path = new File("src/test/resources/marshalled_bytes_8.x.bin").toPath();
```

Comment:
@gustavonalle I've added the sample program and instructions as a comment at the bottom of this class.